cdk/dcm_floodlight/transform.js

```diff
@@ -1,21 +1,27 @@
-/* eslint-disable no-param-reassign */
 const get = require("get-value");
+const _ = require("lodash");
 const {
   removeUndefinedAndNullValues,
-  isDefinedAndNotNull,
-  getFieldValueFromMessage
+  isDefinedAndNotNull
 } = require("rudder-transformer-cdk/build/utils");
-const logger = require("../../logger");
 const {
   CustomError,
   getIntegrationsObj,
   isEmpty,
-  isEmptyObject
+  isEmptyObject,
+  getValueFromPropertiesOrTraits,
+  getHashFromArray
 } = require("../../v0/util");
+const {
+  GENERIC_TRUE_VALUES,
+  GENERIC_FALSE_VALUES
+} = require("../../constants");
+const { BLACKLISTED_CHARACTERS } = require("./config");
 
 // append properties to endpoint
 // eg: ${endpoint}key1=value1;key2=value2;....
-const appendProperties = (endpoint, payload) => {
+const appendProperties = payload => {
+  let endpoint = "";
   endpoint += Object.keys(payload)
     .map(key => {
       return `${key}=${payload[key]}`;
```

Comment:
Should we consider URL encoding these parameters?

Reply:
Yes, DCM expects it to be URL encoded for custom variables. Ref - https://support.google.com/campaignmanager/answer/2823222?hl=en

Reply:
for other parameters, DCM provides us with the values like (cat, src) which do not contain any special characters and some DCM parameters values are restricted to [0|1]